examples_gbs/run_tutorial_similarity.py

```diff
@@ -0,0 +1,286 @@
+# These feature vectors contain information about the graphs and can be viewed as a mapping to a
+# high-dimensional feature space, a technique often used in machine learning that allows us to
+# employ properties of the feature space to separate and classify the vectors.
+#
+# The feature vector of a graph can be composed in a variety of ways. One approach is to build it
+# using statistics from a GBS device configured to sample from the graph, as we now discuss.
+#
+# We begin by defining the concept of an *orbit*, which contains all the GBS samples that are
+# equivalent under permutation of the modes. A sample can be converted to its corresponding orbit
+# using the :func:`~.sample_to_orbit` function. For example, the first sample of ``m0`` is ``[0,
+# 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0]`` and has orbit:
+
+print(similarity.sample_to_orbit(m0[0]))
+
+##############################################################################
+# Here, ``[1, 1]`` means that two photons were detected, each in a separate mode. Other samples
+# can be randomly generated from the ``[1, 1]`` orbit using:
+
+print(similarity.orbit_to_sample([1, 1], modes=m0.modes))
+
+##############################################################################
+# Orbits provide a useful way to coarse-grain the samples from GBS into outcomes that are
+# statistically more likely to be observed. However, we are interested in coarse-graining further
+# into *events*, which correspond to a combination of orbits with the same photon number such
+# that the number of photons counted in each mode does not exceed a fixed value
+# ``max_count_per_mode``. To understand this, let's look at all of the orbits with a photon
+# number of 5:
+
+print(list(similarity.orbits(5)))
+
+##############################################################################
+# All 5-photon samples fall into one of the orbits above. A 5-photon event with
+# ``max_count_per_mode = 3`` means that we include the orbits: ``[[1, 1, 1, 1, 1], [2, 1, 1, 1],
+# [3, 1, 1], [2, 2, 1], [3, 2]]`` and ignore the orbits ``[[4, 1], [5]]``. For example,
+# the sample ``[0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 3, 0, 0, 0]`` is a 5-photon event:
+
+print(similarity.sample_to_event([0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 3, 0, 0, 0], 3))
+
+##############################################################################
+# Samples with more than ``max_count_per_mode`` in any mode are not counted as part of the event:
+
+print(similarity.sample_to_event([0, 4, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0], 3))
+
+##############################################################################
+# Now that we have mastered orbits and events, how can we make a feature vector? It was shown in
+# :cite:`schuld2019quantum` that one way of making a feature vector of a graph is through the
+# probabilities of events. Specifically, for a :math:`k` photon event :math:`E_{k, n_{\max}}`

```

Comment:
Better: frequencies? One may not normalise!

Reply:
Ah that's a good point! Have changed here, but kept "probability(ies)" later on because that's how we coded it up. I do wonder if we should normalise the vectors (resulting in frequencies rather than probabilities), but currently the vectors are directly the probabilities.